cpp/src/io/parquet/writer_impl.cu

```diff
@@ -1042,7 +1059,7 @@ auto build_chunk_dictionaries(hostdevice_2dvector<gpu::EncColumnChunk>& chunks,
          col_desc[chunk.col_desc_id].physical_type == Type::BYTE_ARRAY)) {
       chunk.use_dictionary = false;
     } else {
-      chunk.use_dictionary = true;
+      chunk.use_dictionary = dict_policy != dictionary_policy::NEVER;
```

Comment:
Can we short-circuit `build_chunk_dictionaries` when `dictionary_policy == NEVER`?

Reply:
I'll investigate.  I haven't done the deep dive into `initialize_chunk_hash_maps` and `populate_chunk_hash_maps` to see if there are other things being initialized beyond the dictionaries.

Reply:
I added the short-circuit and it doesn't appear to hurt anything. Still want to look a little closer and make sure.